cmd/ingester/app/flags_test.go

```diff
@@ -49,6 +53,45 @@ func TestOptionsWithFlags(t *testing.T) {
 	assert.Equal(t, kafka.EncodingJSON, o.Encoding)
 }
 
+func TestTLSFlags(t *testing.T) {
+	tests := []struct {
+		flags    []string
+		expected auth.AuthenticationConfig
+	}{
+		{
+			flags:    []string{},
+			expected: auth.AuthenticationConfig{Authentication: "none", Kerberos: auth.KerberosConfig{ServiceName: "kafka", ConfigPath: "/etc/krb5.conf", KeyTabPath: "/etc/security/kafka.keytab"}},
+		},
+		{
+			flags:    []string{"--kafka.consumer.authentication=foo"},
+			expected: auth.AuthenticationConfig{Authentication: "foo", Kerberos: auth.KerberosConfig{ServiceName: "kafka", ConfigPath: "/etc/krb5.conf", KeyTabPath: "/etc/security/kafka.keytab"}},
+		},
+		{
+			flags:    []string{"--kafka.consumer.authentication=kerberos", "--kafka.consumer.tls.enabled=true"},
+			expected: auth.AuthenticationConfig{Authentication: "kerberos", Kerberos: auth.KerberosConfig{ServiceName: "kafka", ConfigPath: "/etc/krb5.conf", KeyTabPath: "/etc/security/kafka.keytab"}, TLS: tlscfg.Options{Enabled: true}},
+		},
+		{
+			flags:    []string{"--kafka.consumer.authentication=tls"},
+			expected: auth.AuthenticationConfig{Authentication: "tls", Kerberos: auth.KerberosConfig{ServiceName: "kafka", ConfigPath: "/etc/krb5.conf", KeyTabPath: "/etc/security/kafka.keytab"}, TLS: tlscfg.Options{Enabled: true}},
+		},
+		{
+			flags:    []string{"--kafka.consumer.authentication=tls", "--kafka.consumer.tls.enabled=false"},
```

Comment:
This means that if someone explicitly set `tls.enabled` to `false, this will be silently changed to `true`. I have nothing against that, but perhaps there's a way to detect whether this value was explicitly provided and keep whatever the user set?

Reply:
it can be done by changing the `enabled` property to a pointer. I am not sure if there is much value in it. Some signatures will have to be also changed to accept the logger.

More controversial is this setting
```
// --kafka.consumer.authentication=kerberos",
--kafka.consumer.tls.enabled=true"
```
Somebody omitting the auth type and specifying `tls.enabled=true`.

Reply:
@yurishkuro any thoughts on this? I don't have a strong opinion whether `kafka.consumer.tls.enabled=true` should set `kafka.consumer.authentication=tls`.

Reply:
From Gitter, we just saw a case where a user wants TLS encryption between the agent and collector without the auth parts. `kafka.producer.tls.enabled=true` + `kafka.producer.authentication=none` is a valid combination.

Reply:
You are right, it's confusing for people to have both `kafka.producer.tls.enabled` and `kafka.producer.authentication` they will probably tend to forget the second one. 

I have updated the PR to set `tls` auth when `.tls.enabled` is `true`.

Reply:
> I have updated the PR to set tls auth when .tls.enabled is true.

It's the opposite. `tls.enabled` does not imply `authentication=tls`. It's perfectly possible to have `authentication=none` but the traffic be encrypted using TLS.

Reply:
@jpkrohling  I am not sure if I understand you...

> It's perfectly possible to have authentication=none but the traffic be encrypted using TLS.

This is exactly what the PR does.

Reply:
Sorry, I misunderstood you, you are absolutely correct :-)